src/base/net/geoipmanager.cpp

```diff
@@ -99,24 +98,40 @@ void GeoIPManager::loadDatabase()
     QString error;
     m_geoIPDatabase = GeoIPDatabase::load(filepath, error);
     if (m_geoIPDatabase)
-        Logger::instance()->addMessage(tr("GeoIP database loaded. Type: %1. Build time: %2.")
+        Logger::instance()->addMessage(tr("IP database loaded. Type: %1. Build time: %2.")
             .arg(m_geoIPDatabase->type(), m_geoIPDatabase->buildEpoch().toString()),
             Log::INFO);
     else
-        Logger::instance()->addMessage(tr("Couldn't load GeoIP database. Reason: %1").arg(error), Log::WARNING);
+        Logger::instance()->addMessage(tr("Couldn't load IP database. Reason: %1").arg(error), Log::WARNING);
 
     manageDatabaseUpdate();
 }
 
 void GeoIPManager::manageDatabaseUpdate()
 {
-    if (!m_geoIPDatabase || (m_geoIPDatabase->buildEpoch().daysTo(QDateTime::currentDateTimeUtc()) >= UPDATE_INTERVAL))
+    const auto expired = [](const QDateTime &testDateTime)
+    {
+        const QDate testDate = testDateTime.date();
+        const QDate curDate = QDateTime::currentDateTimeUtc().date();
+
+        if ((testDate.year() < curDate.year()) && (curDate.day() > 1))
+            return true;
+
+        if ((testDate.month() < curDate.month()) && (curDate.day() > 1))
```

Comment:
Why check for `curDate.day() > 1`?

Also if you want the date difference to be 1 month for `expired()` to be true, you can just write (untested though):
```c++
return ((testDate.year() < curDate.year()) || (testDate.month() < curDate.month()));
```

Reply:
DB-IP updates the free db once per month. It doesn't indicate exactly when. Since we test on UTC time, I would like to wait till the 2nd day of the month before we try to download the new db.

Reply:
But your code will wait one extra day in the case when the database was updated last time in the month before last and earlier (e.g. currently is September 1st and database is updated in July).

Reply:
Look at the generated URL. It will ask for current month. And if current month hasn't yet been uploaded it will fail. It will not be served the last available db.
Unfortunately DB-IP doesn't provide a static URL :(

Reply:
Seems you don't understand me...

Reply:
You can't update at 1st day of month even if you have too old database.

Reply:
My whole point is to not try to update on 1st day of the month (UTC) because we don't know the exact timezone the server operates on nor the exact timeframe in the 1st day that the server will be updated.
Is that a bad "optimization"?
In any case, if you have a too old db (or none at all) and this month's db hasn't been uploaded then you won't be able to update the db anyways.

Reply:
I think using a slightly outdated DB is not a problem since it will be updated eventually. With that said, I would opt for simpler code... but whatever, you can have the final decision.
Also seems DB-IP will keep last month DB URL for some time before they take it down.

Reply:
Ok. Seems you're right.

Reply:
I forgot to mention that I this kinda the result of the API of `DownloadManager`. If `DownloadResult` also propagated the `QNetworkReply::NetworkError` I could use it to fine tune the behavior. I would test for `QNetworkReply::ContentNotFoundError` and try to download the previous month's db (previous from current month). Although undocumented DB-IP seems to still serve January's db if you change the URL.

Can I just add a `QNetworkReply::NetworkError` field into the `DownloadResult` sruct or a more extensive refactoring is required?